docs/markdown/Builtin-options.md

```diff
@@ -53,6 +55,10 @@ accurate, especially on Linux where different distributions have different
 defaults. Using a [cross file](Cross-compilation.md#defining-the-environment),
 particularly the paths section may be necessary.
 
+Both `libpkgconfigdir` and `sharepkgconfigdir` default to `libdata/pkgconfig`
+on FreeBSD, and `<default libdir>/pkgconfig` and `share/pkgconfig` otherwise.
+"default libdir" means the default value of `libdir`.
+
```

Comment:
That doesn't tell why/how to use those options.

As far as I understand this patch, libpkgconfigdir is used by the pkgconfig module as default install dif for .pc files. But I don't see where sharepkgconfigdir gets used.

Reply:
`libpkgconfigdir` is for architecture-dependent library files and `sharepkgconfigdir` is for architecture-independent data files.

Reply:
Ok, thanks for the clarification, that should be told in the doc ;-)

As far as I know we don't support architecture-independent generator yet, I remember there is a PR somewhere... or did that got merged already?

Reply:
I know meson doesn't support generating architecture-independent .pc files. My intention was to allow projects which currently use `configuration_data ` and `configure_file` to generate .pc files to have a way to handle the path properly.

Reply:
#5775 should provide architecture-independent .pc files as far as I can tell.

Reply:
> Ok, thanks for the clarification, that should be told in the doc ;-)

```
| libpkgconfigdir                      | see below     | Pkg-config directory | 
| sharepkgconfigdir                    | see below     | Architecture-independent pkg-config directory |
```

I already add these two rows to the table in the same file. Do you think we need a more detailed explanation in the doc?

Reply:
I'm not sure we should have options about arch indep pc files, until we actually can generate them. I would leave sharepkgconfigdir outside the scope of the PR and add it in the other PR that adds dataonly pc file.